client/state/utils.js

```diff
@@ -24,25 +24,63 @@ export function isValidStateWithSchema( state, schema, checkForCycles = false, b
 	return result.valid;
 }
 
-export function createReducer( initialState = null, handlers = {}, schema = null ) {
-	const defaultHandlers = {
-		[SERIALIZE]: ( state ) => {
-			if ( schema !== null ) {
-				return state;
-			}
+/**
+ * Returns a reducer function with state calculation determined by the result
+ * of invoking the handler key corresponding with the dispatched action type,
+ * passing both the current state and action object. Defines default
+ * serialization (persistence) handlers based on the presence of a schema.
+ *
+ * @param  {*}        initialState    Initial state
+ * @param  {Object}   definedHandlers Object mapping action types to state
+ *                                    action handlers
+ * @param  {?Object}  schema          JSON schema object for deserialization
+ *                                    validation
+ * @return {Function}                 Reducer function
+ */
+export function createReducer( initialState = null, definedHandlers = {}, schema = null ) {
+	// Define default handlers for serialization actions. If no schema is
+	// provided, always return the initial state. Otherwise, allow for
+	// serialization and validate on deserialize.
+	let defaultHandlers;
+	if ( schema ) {
+		defaultHandlers = {
+			[ SERIALIZE ]: ( state ) => state,
+			[ DESERIALIZE ]: ( state ) => {
+				if ( isValidStateWithSchema( state, schema ) ) {
+					return state;
+				}
 
-			return initialState;
-		},
-		[DESERIALIZE]: ( state ) => {
-			if ( schema !== null && isValidStateWithSchema( state, schema ) ) {
-				return state;
+				return initialState;
 			}
+		};
+	} else {
+		defaultHandlers = {
+			[ SERIALIZE ]: () => initialState,
+			[ DESERIALIZE ]: () => initialState
+		};
+	}
 
-			return initialState;
-		}
+	const handlers = {
+		...defaultHandlers,
+		...definedHandlers
```

Comment:
Ah, I see what you mean. Maybe `fallbackHandlers` instead of `defaultHandlers`? Or `customHandlers` instead of `definedHandlers`? Or `modifiedHandlers` instead of `handlers` (changing `definedHandlers` back to `handlers`)?